fendermint/actors/machine/src/lib.rs

```diff
@@ -68,9 +70,12 @@ pub trait MachineActor {
 
         let id_addr = to_id_address(rt, params.address, false)?;
 
-        rt.transaction(|st: &mut Self::State, _| st.init(id_addr))?;
+        let kind = rt.transaction(|st: &mut Self::State, _| {
+            st.init(id_addr)?;
+            Ok(st.kind())
+        })?;
 
-        emit_evm_event(rt, machine_initialized(id_addr))
+        emit_evm_event(rt, machine_initialized(kind as u8, id_addr))
```

Comment:
rust enums auto-map to ints... and solidity events can't have enums... so just using `u8` here.

0 = bucket
1 = timehub